tests/integration/clientv3/examples/example_watch_test.go

```diff
@@ -12,89 +12,146 @@
 }
 
+func mockWatcher_watchWithRange() {
+	fmt.Println(`PUT "foo1" : "bar1"`)
+	fmt.Println(`PUT "foo2" : "bar2"`)
+	fmt.Println(`PUT "foo3" : "bar3"`)
+}
+
 func ExampleWatcher_watchWithRange() {
-	cli, err := clientv3.New(clientv3.Config{
-		Endpoints:   endpoints,
-		DialTimeout: dialTimeout,
-	})
-	if err != nil {
-		log.Fatal(err)
-	}
-	defer cli.Close()
-
-	// watches within ['foo1', 'foo4'), in lexicographical order
-	rch := cli.Watch(context.Background(), "foo1", clientv3.WithRange("foo4"))
-	for wresp := range rch {
-		for _, ev := range wresp.Events {
-			fmt.Printf("%s %q : %q\n", ev.Type, ev.Kv.Key, ev.Kv.Value)
+	forUnitTestsRunInMockedContext(mockWatcher_watchWithRange, func() {
+		cli, err := clientv3.New(clientv3.Config{
+			Endpoints:   exampleEndpoints(),
+			DialTimeout: dialTimeout,
+		})
+		if err != nil {
+			log.Fatal(err)
 		}
-	}
-	// PUT "foo1" : "bar"
-	// PUT "foo2" : "bar"
-	// PUT "foo3" : "bar"
+		defer cli.Close()
+
+		// watches within ['foo1', 'foo4'), in lexicographical order
+		rch := cli.Watch(context.Background(), "foo1", clientv3.WithRange("foo4"))
+
+		go func() {
+			cli.Put(context.Background(), "foo1", "bar1")
+			cli.Put(context.Background(), "foo5", "bar5")
+			cli.Put(context.Background(), "foo2", "bar2")
+			cli.Put(context.Background(), "foo3", "bar3")
+		}()
+
+		i := 0
+		for wresp := range rch {
+			for _, ev := range wresp.Events {
+				fmt.Printf("%s %q : %q\n", ev.Type, ev.Kv.Key, ev.Kv.Value)
+				i++
+				if i == 3 {
+					// After 3 messages we are done.
+					cli.Delete(context.Background(), "foo", clientv3.WithPrefix())
+					cli.Close()
+					return
+				}
+			}
+		}
+	})
+
+	// Output:
+	// PUT "foo1" : "bar1"
+	// PUT "foo2" : "bar2"
+	// PUT "foo3" : "bar3"
+}
+
+func mockWatcher_watchWithProgressNotify() {
+	fmt.Println(`wresp.Header.Revision: 0`)
+	fmt.Println(`wresp.IsProgressNotify: false`)
 }
 
 func ExampleWatcher_watchWithProgressNotify() {
-	cli, err := clientv3.New(clientv3.Config{
-		Endpoints:   endpoints,
-		DialTimeout: dialTimeout,
+	forUnitTestsRunInMockedContext(mockWatcher_watchWithProgressNotify, func() {
+		cli, err := clientv3.New(clientv3.Config{
+			Endpoints:   exampleEndpoints(),
+			DialTimeout: dialTimeout,
+		})
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		rch := cli.Watch(context.Background(), "foo", clientv3.WithProgressNotify())
+		go func() {
+			time.Sleep(time.Second)
+			err := cli.Close()
+			if err != nil {
+				log.Fatal(err)
+			}
+		}()
+		wresp := <-rch
+		fmt.Printf("wresp.Header.Revision: %d\n", wresp.Header.Revision)
+		fmt.Println("wresp.IsProgressNotify:", wresp.IsProgressNotify())
 	})
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	rch := cli.Watch(context.Background(), "foo", clientv3.WithProgressNotify())
-	wresp := <-rch
-	fmt.Printf("wresp.Header.Revision: %d\n", wresp.Header.Revision)
-	fmt.Println("wresp.IsProgressNotify:", wresp.IsProgressNotify())
+
+	// TODO: Rather wresp.IsProgressNotify: true should be expected
```

Comment:
Probably need to configure `WatchProgressNotifyInterval` to a small value when creating the lazy cluster.

ref:
https://github.com/etcd-io/etcd/blob/0aab02e7b5ab94ef666392f7a56e0780e7cb27f5/clientv3/integration/watch_test.go#L587

Reply:
Fixed.